[Bug] `pyarrow` not found, install `pyarrow>=3.0.0`

### Is this a new bug in dbt-bigquery?

- [X] I believe this is a new bug in dbt-bigquery
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

CI failure: https://github.com/dbt-labs/dbt-bigquery/actions/runs/8459554085/job/23176082535

### Expected Behavior

No CI failure

### Steps To Reproduce

https://github.com/dbt-labs/dbt-bigquery/actions/runs/8459554085/job/23176082535

### Relevant log output

```shell
try:
            import pyarrow
        except ImportError as exc:  # pragma: NO COVER
            if raise_if_error:
>               raise exceptions.LegacyPyarrowError(
                    "pyarrow package not found. Install pyarrow version >="
                    f" ***_MIN_PYARROW_VERSION***."
                ) from exc
E               google.cloud.bigquery.exceptions.LegacyPyarrowError: pyarrow package not found. Install pyarrow version >= 3.0.0.
```


### Environment

```markdown
- OS: all
- Python: all
- dbt-core: 1.8
- dbt-bigquery: 1.8
```


### Additional Context

_No response_

--- a/.changes/unreleased/Fixes-20240327-210249.yaml
+++ b/.changes/unreleased/Fixes-20240327-210249.yaml
@@ -1,0 +1,7 @@
+kind: Fixes
+body: Add `pandas` extra for `google-cloud-bigquery` to pick up missing `pyarrow`
+  dependency
+time: 2024-03-27T21:02:49.619691-04:00
+custom:
+  Author: mikealfare
+  Issue: "1152"

--- a/setup.py
+++ b/setup.py
@@ -54,7 +54,7 @@ setup(
         "dbt-common>=0.1.0a1,<2.0",
         "dbt-adapters>=0.1.0a1,<2.0",
         # 3.20 introduced pyarrow>=3.0 under the `pandas` extra
-        "google-cloud-bigquery[pandas]>=3.0",
+        "google-cloud-bigquery[pandas]>=3.0,<4.0",
         "google-cloud-storage~=2.4",
         "google-cloud-dataproc~=5.0",
         # ----

--- a/setup.py
+++ b/setup.py
@@ -53,7 +53,8 @@ setup(
     install_requires=[
         "dbt-common>=0.1.0a1,<2.0",
         "dbt-adapters>=0.1.0a1,<2.0",
-        "google-cloud-bigquery~=3.0",
+        # 3.20 introduced pyarrow>=3.0 under the `pandas` extra
+        "google-cloud-bigquery[pandas]>=3.0",
         "google-cloud-storage~=2.4",
         "google-cloud-dataproc~=5.0",
         # ----

